src/main/scala/translating/ILtoIL.scala

```diff
@@ -0,0 +1,279 @@
+    program ++= ", "
+    program ++= "Condition("
+    node.condition = node.condition.map(visitExpr)
+    program ++= ")" // Condition
+    program ++= ")" // GoTo
+    node
+  }
+
+  override def visitDirectCall(node: DirectCall): Jump = {
+    program ++= "DirectCall("
+    program ++= procedureIdentifier(node.target)
+    program ++= ", "
+    program ++= "Condition("
+    node.condition = node.condition.map(visitExpr)
+    program ++= ")" // Condition
+    program ++= ")" // DirectCall 
+    node
+  }
+
+  override def visitIndirectCall(node: IndirectCall): Jump = {
+    program ++= "IndirectCall("
+    node.target = visitVariable(node.target)
+    program ++= ", "
+    program ++= "Condition("
+    node.condition = node.condition.map(visitExpr)
+    program ++= ")" // Condition
+    program ++= ")" // IndirectCall 
+    node
+  }
+
+  override def visitBlock(node: Block): Block = {
+    program ++= getIndent()
+    program ++= "Block(" + blockIdentifier(node) + ",\n"
+    indentLevel += 1
+    program ++= getIndent()
+    program ++= "statements(\n"
+    indentLevel += 1
+
+    for (i <- node.statements.indices) {
+      program ++= getIndent()
+      node.statements(i) = visitStatement(node.statements(i))
+      program ++= "\n"
+    }
+    indentLevel -= 1
+    program ++= getIndent() + "),\n"
+    program ++= getIndent() + "jumps(\n"
+    indentLevel += 1
+    for (i <- node.jumps.indices) {
+      program ++= getIndent()
+      node.jumps(i) = visitJump(node.jumps(i))
+      program ++= "\n"
+    }
+    indentLevel -= 1
+    program ++= getIndent() + ")\n"
+    indentLevel -= 1
+    program ++= getIndent()
+    program ++= ")\n"
+    node
+  }
+
+  override def visitProcedure(node: Procedure): Procedure = {
+    program ++= "Procedure(" + procedureIdentifier(node) + ", "
+    indentLevel += 1
+  
+      program ++= "in("
+    for (i <- node.in.indices) {
+      node.in(i) = visitParameter(node.in(i))
+      if (i != node.in.size - 1)
+        program ++= ", "
+    }
+    program ++= "), "
+    program ++= "out("
+    for (i <- node.out.indices) {
+      node.out(i) = visitParameter(node.out(i))
+      if (i != node.out.size - 1)
+        program ++= ", "
+    }
+    program ++= "), "
+    program ++= "blocks(\n"
+    for (i <- node.blocks.indices) {
+      node.blocks(i) = visitBlock(node.blocks(i))
+    }
+    program ++= "))\n"
+    indentLevel -= 1
+    node
+  }
+
+  override def visitParameter(node: Parameter): Parameter = {
+    program ++= "Parameter("
+    node.value = visitRegister(node.value)
+    program ++= ")"
+    node
+  }
+
+  override def visitProgram(node: Program): Program = {
+    for (i <- node.procedures.indices) {
+      val updatedProcedure = visitProcedure(node.procedures(i))
+      val targetProcedure = node.procedures(i)
+      if (targetProcedure == node.mainProcedure) {
+        node.mainProcedure = updatedProcedure
+      }
+      node.procedures(i) = updatedProcedure
+    }
+    node
+  }
```

Comment:
You don't need to copy in methods from the base class that haven't been changed, but using this version which is designed for mutating the IR is incorrect.